src/group_theory/group_action.lean

```diff
@@ -131,35 +136,51 @@ noncomputable def orbit_equiv_quotient_stabilizer (b : β) :
 equiv.symm (@equiv.of_bijective _ _
   (λ x : quotient (stabilizer α b), quotient.lift_on' x
     (λ x, (⟨x • b, mem_orbit _ _⟩ : orbit α b))
-    (λ g h (H : _ = _), subtype.eq $ (group_action.bijective (g⁻¹)).1
+    (λ g h (H : _ = _), subtype.eq $ (mul_action.bijective (g⁻¹)).1
       $ show g⁻¹ • (g • b) = g⁻¹ • (h • b),
-      by rw [← monoid_action.mul, ← monoid_action.mul,
-        H, inv_mul_self, monoid_action.one]))
+      by rw [← mul_action.mul_smul, ← mul_action.mul_smul,
+        H, inv_mul_self, mul_action.one_smul]))
 ⟨λ g h, quotient.induction_on₂' g h (λ g h H, quotient.sound' $
   have H : g • b = h • b := subtype.mk.inj H,
   show (g⁻¹ * h) • b = b,
-  by rw [monoid_action.mul, ← H, ← monoid_action.mul, inv_mul_self, monoid_action.one]),
+  by rw [mul_action.mul_smul, ← H, ← mul_action.mul_smul, inv_mul_self, mul_action.one_smul]),
   λ ⟨b, ⟨g, hgb⟩⟩, ⟨g, subtype.eq hgb⟩⟩)
 
 end
 
-open quotient_group  monoid_action is_subgroup
+open quotient_group mul_action is_subgroup
 
 def mul_left_cosets (H : set α) [is_subgroup H]
   (x : α) (y : quotient H) : quotient H :=
 quotient.lift_on' y (λ y, quotient_group.mk ((x : α) * y))
   (λ a b (hab : _ ∈ H), quotient_group.eq.2
     (by rwa [mul_inv_rev, ← mul_assoc, mul_assoc (a⁻¹), inv_mul_self, mul_one]))
 
-instance (H : set α) [is_subgroup H] : group_action α (quotient H) :=
+instance (H : set α) [is_subgroup H] : mul_action α (quotient H) :=
 { smul := mul_left_cosets H,
-  one := λ a, quotient.induction_on' a (λ a, quotient_group.eq.2
+  one_smul := λ a, quotient.induction_on' a (λ a, quotient_group.eq.2
     (by simp [is_submonoid.one_mem])),
-  mul := λ x y a, quotient.induction_on' a (λ a, quotient_group.eq.2
+  mul_smul := λ x y a, quotient.induction_on' a (λ a, quotient_group.eq.2
     (by simp [mul_inv_rev, is_submonoid.one_mem, mul_assoc])) }
 
 instance mul_left_cosets_comp_subtype_val (H I : set α) [is_subgroup H] [is_subgroup I] :
-  group_action I (quotient H) :=
-group_action.comp_hom (quotient H) (subtype.val : I → α)
+  mul_action I (quotient H) :=
+mul_action.comp_hom (quotient H) (subtype.val : I → α)
+
+end mul_action
+
+/-- Typeclass for multiplicative actions on additive structures. This generalizes group modules. -/
+class mul_action_add (α : Type u) (β : Type v) [monoid α] [add_monoid β] extends mul_action α β :=
+(smul_add : ∀(r : α) (x y : β), r • (x + y) = r • x + r • y)
```

Comment:
I think swapping lhs and rhs would make a better simp lemma

Reply:
That goes contrary to how these distributivity laws are written elsewhere. In general I think neither direction is good enough to be a simp-lemma. You rewrite both ways quite often (I think).

Reply:
The difference here is that `r` and `x, y` have different types. Reversing the sides puts `x` and `y` together and then they can interact in some way.

Reply:
Sure, but very often you actually don't want them together, in practice. There is only one occurence of `← smul_add` and two `(smul_add _ _ _).symm` in mathlib at the moment. But there are lots of occurences with the current direction.

Reply:
Neither direction should be a simp lemma, because distributivity expansion can blow up theorems, but as a manually applied rewrite it should be this direction, because the LHS is more likely to match than the RHS (because you are duplicating a term rather than merging terms).